frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx

```diff
@@ -430,8 +430,11 @@ const flatten = ({
         isDefault: true,
         name: defaultSource.name,
         namespace: DEFAULT_SOURCE_NAMESPACE,
+        // Prevent label filter crash when catalogSource doesn't exist
+        metadata: {},
```

Comment:
> I think an easier solution for the name filter problem would be to just change the default name filter in `public/components/factory/table-filters.ts` to something like this:
> 
> ```
>   name: (filter, obj) => fuzzyCaseInsensitive(filter, obj?.metadata?.name ?? obj?.name),
> ```
> 
> There are a few other custom name filters defined in table filters because of a similar issue where the name property isn't in metadata. Maybe we can add a TODO or FIXME comment to those to say that the default name filter will work for these resources after this update.

This suggestion would allow us to remove the usage of `catalog-source-name` but there are other table-filters that filter on other properties that are not directly under those two fields. For example `projects`, they use  annotation to store the display name shown to the user, so the solution you’ve proposed will work for `catalogSource` but not for everything, textFilter is unavoidable.
But we can update it to your suggestion so that we can avoid `table-filters` which share the logic you proposed.

Reply:
ack